url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/ui/LinuxOptionPanel.java

```diff
@@ -0,0 +1,204 @@
+        txtInitrd.setMinimumSize(dim);
+        JButton chooseInitrd = new JButton("...");
+        chooseInitrd.addActionListener(listenButton);
+        chooseInitrd.setActionCommand("chooseInitrd");
+
+        txtCmdline = new JTextField(opts.getCommandLine());
+        txtCmdline.setPreferredSize(dim);
+        txtCmdline.setMinimumSize(dim);
+
+        //コンポーネントを配置
+        GridBagLayout layout = new GridBagLayout();
+        int gy = 0;
+        setLayout(layout);
+
+        GridBagLayoutHelper.add(this, layout, new JLabel("Device Tree Image:", SwingConstants.RIGHT),
+                0, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, txtDeviceTree,
+                1, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, chooseDeviceTree,
+                2, gy, GridBagConstraints.RELATIVE, 1);
+
+        gy += 1;
+        GridBagLayoutHelper.add(this, layout, chkUseDeviceTree,
+                1, gy, 3, 1);
+
+        gy += 1;
+        GridBagLayoutHelper.add(this, layout, new JLabel("Kernel Image:", SwingConstants.RIGHT),
+                0, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, txtImage,
+                1, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, chooseKernel,
+                2, gy, GridBagConstraints.RELATIVE, 1);
+
+        gy += 1;
+        GridBagLayoutHelper.add(this, layout, new JLabel("Initrd Image:", SwingConstants.RIGHT),
+                0, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, txtInitrd,
+                1, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, chooseInitrd,
+                2, gy, GridBagConstraints.RELATIVE, 1);
+
+        gy += 1;
+        GridBagLayoutHelper.add(this, layout, new JLabel("Command line:", SwingConstants.RIGHT),
+                0, gy, 1, 1);
+        GridBagLayoutHelper.add(this, layout, txtCmdline,
+                1, gy, 2, 1);
+
+        setBorder(BorderFactory.createTitledBorder("Linux Boot Options"));
+    }
+
+    /**
+     * Linux 起動時に渡すオプションを取得します。
+     *
+     * @return エミュレータで Linux 起動時に渡すオプション
+     */
+    public LinuxOption getOption() {
+        updateOption();
+
+        return opts;
+    }
+
+    /**
+     * パネル内の各コンポーネントの設定値をオプションオブジェクトに反映させます。
+     *
+     * 設定値が URI として（ホスト名の場合）解釈できない場合は、
+     * オプションオブジェクトの値は変更されません。
+     */
+    protected void updateOption() {
+        try {
+            opts.setDeviceTreeImage(new URI(txtDeviceTree.getText()));
+            opts.setKernelImage(new URI(txtImage.getText()));
+            opts.setInitrdImage(new URI(txtInitrd.getText()));
+            opts.setCommandLine(txtCmdline.getText());
+        } catch (URISyntaxException e) {
+            //ignored
+        }
+    }
+
+    class ButtonListener implements ActionListener {
+        LinuxOptionPanel parent;
+
+        public ButtonListener(LinuxOptionPanel c) {
+            parent = c;
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 183-201 into a separate method.